examples/async_fl/async_aggregator.py

```diff
@@ -55,13 +73,23 @@ def tictak_client_tasks(self, sampled_clients, num_clients_to_collect):
                     exe_cost['communication']
                 # if the client is not active by the time of collection, we consider it is lost in this round
                 start_time += constant_checkin_period
-                if self.client_manager.isClientActive(client_to_run, roundDuration + start_time):
+                end_time = roundDuration + start_time
+                end_list.append( end_time )
+                while start_time > end_list[end_j] :
+                    concurreny_count -= 1
+                    end_j += 1
+                if concurreny_count > self.max_concurrency:
+                    end_list.pop()
+                    continue
```

Comment:
By breaking the loop here, are we limiting the number of participants for each model version to max_concurrency?

For example, if we have an aggregation target of 40, and the max concurrency is set to 20, would round_completion_handler() every be called and new clients be sampled?

Reply:
I think that would not be a problem. Here I just make sure the concurrent number of participants <= 20, while it can still launch > 40 clients in a larger time window.  `continue` does not break the loop, instead jump to the next loop.

(though it depends on the traces, so I need to set a larger over [here](https://github.com/SymbioticLab/FedScale/blob/e8bfa5cdb2662bd752e0053196cad1a65000cf2a/examples/async_fl/async_aggregator.py#L193))

Reply:
I see, thanks. It seems like num_participants is no longer used in the code or did I miss it?

Reply:
Yes. I have removed it.